qiskit/pulse/library/pulse.py

```diff
@@ -47,6 +48,12 @@ def __init__(
         self.duration = duration
         self.name = name
         if limit_amplitude is not None:
+            warnings.warn(
+                "Instantiating a pulse instance with 'limit_amplitude' has been deprecated. "
+                f"Please directly update class variable '{self.__class__.__name__}.limit_amplitude.'",
+                DeprecationWarning,
+                stacklevel=4,
+            )
```

Comment:
Would it not be more flexible to deprecate the class variable instead and do this validation from one instance to the next? With this PR you either validate all your pulses or none of them. On the other hand, I'm not sure if we need the flexibility.

Reply:
Yeah that's the point I'd like to hear your thought. I think instance wise validation is not practical from viewpoint of hardware, because we cannot dynamically change representation in the same channel. I `limit_amplitude` is more like a instance variable of channel rather than pulse.
https://github.com/Qiskit/qiskit-terra/pull/7948#issuecomment-1102019646